### PR TITLE
Migrate selector tests to pytest

### LINKED ISSUE #9868: [SPIKE+] Write a unittest for a simple selection
Creating this one as a Spike + because there is not a single unit test for selection, and selection is very involved.
By trying to write out one unit test for a simple selection, we aim to get a sense of
- what fixture do we need to properly start writing unit tests for selection
- how much effort it takes to refactor selection out of task into it's own module. So task can start with a selected set of nodes to run.

### PR DESCRIPTION
**Pull request by QMalcolm**
resolves #9868 

### Problem

In #9868 we wanted to ensure we write, and can write, some selector tests. After creating the ticket, we actually found some selector tests while we were reorganizing tests. Thus #9868 became about converting these tests to use pytest.

### Solution

Converted selector tests to use pytest, showing that we can write selector tests

### Checklist

- [X] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [X] I have run this code in development and it appears to resolve the stated issue  
- [X] This PR includes tests, or tests are not required/relevant for this PR
- [X] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX
- [X] This PR includes [type annotations](https://docs.python.org/3/library/typing.html) for new and modified functions

### CODE REVIEW

File: tests/unit/graph/test_selector.py [hunk 1]
```diff
@@ -82,7 +68,7 @@ def graph():
 
 
 @pytest.fixture
-def manifest(graph):
+def mock_manifest(graph):
     return _get_manifest(graph)
```
Comment (ChenyuLInx) [inline tests/unit/graph/test_selector.py:72]: Should we be more specific here that we are only mocking the graph functionality part?

File: tests/unit/graph/test_selector.py [hunk 2]
```diff
@@ -231,350 +217,82 @@ def test_invalid_specs(invalid):
...
-    def test__two_models_package_ref(self):
-        self.use_models(
-            {
-                "model_one": "select * from events",
-                "model_two": "select * from {{ref('test_models_compile', 'model_one')}}",
-            }
-        )
+        expected_nodes = [model.unique_id for model in models]
+        assert linker.nodes() == set(expected_nodes)
+        assert list(linker.edges()) == [tuple(expected_nodes)]
```
Comment (ChenyuLInx) [inline tests/unit/graph/test_selector.py:247]: Nitty Nit: It is better to just write out the actual values in the test if it is not too complex. This way reader can understand this edges will yield a list of string without knowing what `model.unique_id`(which is the implementation detail of `model`) is.

File: tests/unit/utils/manifest.py [hunk 3]
```diff
@@ -963,25 +968,58 @@ def macros(
 
 
 @pytest.fixture
-def unit_tests(unit_test_table_model) -> list:
+def unit_tests(unit_test_table_model) -> List[UnitTestDefinition]:
```
Comment (ChenyuLInx) [inline tests/unit/utils/manifest.py:971]: Thanks for all the type annotation!

**Review by ChenyuLInx — APPROVED** [review]
Have two optional nits otherwise looks good to me!

File: tests/unit/graph/test_selector.py (hunk 1 above)
Comment (QMalcolm) [reply tests/unit/graph/test_selector.py:72]: Yea I can change the name to something like `manifest_with_mocked_graph`.

File: tests/unit/graph/test_selector.py (hunk 2 above)
Comment (QMalcolm) [reply tests/unit/graph/test_selector.py:247]: Would a compromise of adding a type hint be sufficient? For example:
```python
expected_nodes: List[str] = [model.unique_id for model in models]
```

My issue with hand writing it out is that I find that it doesn't scale well and makes things more brittle. From a perspective of making it easier to write tests, I think this approach will suit us better.

Comment (ChenyuLInx) [reply tests/unit/graph/test_selector.py:247]: I should have added this point in the last comment.
So in some way, we are writing out some form of implementation in the tests(it is fairly small in this case). IMHO unit tests should be defining input and output explicitly without (or as little as possible) dependencies between them.
I don't think writing out an explicit value makes writing tests harder.

This is not a very strongly held opinion and I am happy that you make the final call here. We can always adjust later on if we find out more.

**Pull request merged**